nucleof072rb/Core/Src/main.c

```diff
@@ -44,12 +47,18 @@
 /* Private variables ---------------------------------------------------------*/
 
 /* USER CODE BEGIN PV */
-
+uint8_t pT_data[3] = {0x01, 0x80, 0x00};
+uint8_t pR_data[3];
+uint16_t size = 3;
+uint32_t timeout = HAL_MAX_DELAY;
 /* USER CODE END PV */
 
 /* Private function prototypes -----------------------------------------------*/
 void SystemClock_Config(void);
 /* USER CODE BEGIN PFP */
+double ADCToPWM(int ADC_val);
```

Comment:
this function is not needed. just directly use the math it wraps up the one time you call it.